### ISSUE TITLE
If a unit dies while working, the number of workers at its structures will not be decreased

### ISSUE THREAD

**Issue by alice-i-cecile** [Leafwing-Studios/Emergence#595]
Currently we're only removing workers when their actions are complete.

When despawning units, we should check their current action and decrement accordingly.

**Comment by TimJentzsch** [comment]
This also sounds like a perfect candidate for a test to make sure that this doesn't happen again

**Comment by alice-i-cecile** [comment]
Yep, absolutely. This would make a really nice integration test.

**Comment by alice-i-cecile** [comment]
#585 is related: we should probably have a custom despawn command too.
